tests/test_measureApCorr.py

```diff
@@ -153,16 +153,18 @@ def testTooFewSources(self):
         """ If there are too few sources, check that an exception is raised."""
         # Create an empty catalog with no sources to process.
         catalog = afwTable.SourceCatalog(self.schema)
-        with self.assertLogs(level=logging.WARNING) as cm:
-            with self.assertRaisesRegex(measureApCorr.MeasureApCorrError, "failed on required algorithm"):
-                self.meas_apCorr_task.run(catalog=catalog, exposure=self.exposure)
-        self.assertIn("Unable to measure aperture correction for required algorithm", cm.output[0])
+        with self.assertRaisesRegex(measureApCorr.MeasureApCorrError,
+                                    "Unable to measure aperture correction for 'test1Ap'"):
+            self.meas_apCorr_task.run(catalog=catalog, exposure=self.exposure)
 
         # We now try again after declaring that the aperture correction is
-        # allowed to fail. This should run cleanly without raising an exception.
+        # allowed to fail. This should run without raising an exception, but
+        # will log a warning.
```

Comment:
This is a really nice example of the utility of this new exception handling option! However, it's not clear to me how a user would set this at runtime, do you have an example for how to do that available somewhere?

Reply:
This is specifically for the aperture correction algorithm: `allowFailure` is the config option that sets which measurements are allowed to fail. It's a feature that DRP uses, but I don't know anything about it. I'm just making the existing tests support the new exception system.

Reply:
Right, I know this is a specific situation where you now use `allowFailure`, and thus here lies the one place where all of this began to make sense to me. 😄

Reply:
No, `allowFailure` has always been in MeasureApCorr, it has nothing to do with my changes. All the change here does is make the test pass, since the details of the exception have changed.